### DISCUSSION TITLE
Add a delegating handler to IHttpClientFactory.createClient()

### DISCUSSION THREAD

**Discussion by ankituipath** [dotnet/aspnetcore#24610] [General]
I wanted to know if there is a feature planned to support adding a delegating handler to IHttpClientFactory.createClient() method.
Currently it only takes a name. 
There is a workaround that we found which was by using IHttpMessageHandlerBuilderFilter and injecting an handler. Our handler takes a few parameters in its constructor and we had to jump through multiple hoops to pass this information at the handler, so that when we request a new client using the factory the handler is included in the pipeline with the right set of information. 
If we had the functionality to specify the Delegating handlers in the createClient (just like we had in HttpClientFactory.Create() of asp.net) method that would have made our life much much easier. Right now the code is not clean, not readable and not maintainable.
Thoughts?

**Comment by davidfowl** [comment]
You wouldn't do it in create client, as it's part of client configuration. There's no way currently to flow arguments from the `CreateClient` call (there are none) to construction of the handler. Can you show what you're currently doing?
